pkg/storage/bloom/v1/reader.go

```diff
@@ -31,6 +32,33 @@ func (r *ByteReader) Blooms() (io.ReadSeeker, error) {
 	return bytes.NewReader(r.blooms.Bytes()), nil
 }
 
+func (r *ByteReader) TarEntries() (Iterator[TarEntry], error) {
```

Comment:
I'm not sure why this needs to return an `Interator` and not just a slice, because you don't build the `TarEntry`s lazily anyway.